packages/rocketchat-user-data-download/server/lib/requests.js

```diff
@@ -0,0 +1,62 @@
+/* globals FileUpload, WebApp */
+
+import fs from 'fs';
+
+WebApp.connectHandlers.use(`${ __meteor_runtime_config__.ROOT_URL_PATH_PREFIX }/user-data-download/`, function(req, res) {
+	const match = /^\/(.*)\/?/.exec(req.url);
+	if (match[1]) {
+		const userId = match[1];
+		const user = RocketChat.models.Users.findOneById(userId);
+		const userName = user ? user.name : userId;
+
+		const operation = RocketChat.models.ExportOperations.findLastOperationByUser(userId);
+
+		if (operation && operation.status === 'completed') {
+			if (operation.generatedFile !== null) {
+
+				if (!Meteor.settings.public.sandstorm && !FileUpload.requestCanAccessFiles(req)) {
+					res.writeHead(403);
+					return res.end();
+				}
+
+				const filePath = operation.generatedFile;
```

Comment:
We are making a lot of assumptions about long term storage being available on the disk.  Could we store the file on the file store provider already setup on the server?

Let's just imagine that this is in a docker container.  If you write the file and then the container is restarted the file is gone.  Mounting a directory is possible... But in some setups this is pretty difficult

Reply:
I've added GridFS, Amazon and Google Storage as options for the location of the generated file.